src/catalua.cpp

```diff
@@ -1238,13 +1278,22 @@ int call_lua( std::string ) {
 }
 // Implemented in mapgen.cpp:
 // int lua_mapgen( map *, std::string, mapgendata, int, float, const std::string & )
-void lua_callback( const char * )
-{
-}
+
 void lua_loadmod( std::string, std::string )
 {
 }
 void game::init_lua()
 {
 }
+
+void lua_callback( const char*, const CallbackArgumentContainer& )
+{
+}
+void lua_callback( const char* )
+{
+}
+
 #endif
+
+void lua_callback( const char*, const CallbackArgumentContainer& );
+void lua_callback( const char* );
```

Comment:
Those two functions are already declared in the header, no need to repeat that declaration here, especially as it's at the end of the cpp file and nothing would make use of it.

Reply:
These two are required for non-Lua builds, IIRC.

Reply:
These are *declarations* (not definitions) and those two functions are already declared in the header. And they are *defined* (for Lua and non-Lua builds) above, and because a definition is implicitly also a declaration, they are declared there as well.

These declarations are indeed needed in the header, but this is not the header.